public/docs/ts/latest/cookbook/graphql.jade

```diff
@@ -0,0 +1,688 @@
+
+.alert.is-important
+  :marked
+    Uri: Can you explain why the GraphQL API makes using things like Firebase different? 
+    There's a "but" in that sentence, but I don't know what it implies.
+    :Kapunahele - the word `but` was wrong. fixed and edited a bit
+
+:marked
+  In order to create a GraphQL schema, you need the `graphql-tools` library.
+  It allows you to write a GraphQL schema as a string and make it executable. 
+  In a terminal window, issue the following command:
+
+code-example(language="sh" class="code-shell").
+    npm install graphql-tools --save
+
+:marked
+  Next, create a file called `in-memory-graphql.ts` in the `app` directory 
+  and paste in the following schema:
+
++makeExample('heroes-graphql/ts/src/app/in-memory-graphql.ts', 'graphql-schema', 'in-memory-graphql.ts')
+
+.alert.is-important
+  :marked
+    Uri: Could you change the use of "we" in the above comments to "you" or the imperative (command without 
+    "you")? :Kapunahele - done.
+.l-sub-section
+  :marked
+    While the schema includes the major points covered in this cookbook, 
+    you can read more in the [GraphQL.org Introduction to GraphQL](http://graphql.org/learn/).
+:marked
+  Now, create your in-memory data:
++makeExample('heroes-graphql/ts/src/app/in-memory-graphql.ts', 'heroes-array', 'in-memory-graphql.ts')
+:marked
+  The next step is writing a server that _resolves_ 
+  the queries from the client based on the schema. 
+  Hence, the GraphQL server consists of _resolver 
+  functions_ that correspond to the _types_ of the schema.
+
+  To create the resolvers, copy the following code and add it to `in-memory-graphql.ts`.
++makeExample('heroes-graphql/ts/src/app/in-memory-graphql.ts', 'resolvers', 'in-memory-graphql.ts')
+
+.l-sub-section
+  :marked
+    For the full explanation about how GraphQL resolvers work see
+    [Execution](http://graphql.org/learn/execution/) on [GraphQL.org](http://graphql.org/).
+
+:marked
+  You may have noticed some functions from `lodash` such as `lodashFind()` so don't 
+  forget to install them from npm and import them:
+code-example(language="sh" class="code-shell").
+    npm install lodash --save  
++makeExample('heroes-graphql/ts/src/app/in-memory-graphql.ts', 'import-lodash', 'in-memory-graphql.ts (imports)')
+
+.alert.is-important
+  :marked
+    Uri: What is the npm install command for the lodash functions? 
+    Do we install the whole lodash library or just the functions? 
+    (I've never used lodash before).
+    :Kapunahele - fixed. for ease of use I've installed all of lodash
+
+
+:marked
+  Notice that the server includes functions that correspond to each 
+  type in the schema _and_ the mutations.
+
+  This mechanism makes writing simple GraphQL servers straightforward&mdash;you simply 
+  resolve a specific type of data. 
+  This removes the coupling between frontend and backend because if you don't need to know the specific
+  query the client makes to create the server Implementation.
```

Comment:
>...between _the_ frontend and backend because ~~if~~ you don't  need to know the specific query the client  makes to create the server implementation.

Add "the", delete "if", lowercase "implementation".